tests/fsharp/Compiler/CompilerAssert.fs

```diff
@@ -217,3 +217,23 @@ module CompilerAssert =
                     Assert.AreEqual(expectedErrorMessage, errorMessage)
                 )
         
+    let ParseWithErrors (source: string) expectedParseErrors = 
+        lock gate <| fun () ->
```

Comment:
If you use `ParseFile` then this may not be necessary, since all parsing operations are thread-safe

Reply:
Yep, I just got back and tried it out (with ``FSharpParsingOptions.Default`` as the fsharp parser opts, which I hope is OK). I'll back out the test dupes @KevinRansom's comment and sort the merge issues and hopefully all will be grand